src/lib/Libwin/dir.c

```diff
@@ -71,20 +72,23 @@ opendir(const char *name)
 	strncat(search, "/*", _MAX_PATH);
 
 	hdir = FindFirstFile(search, &data);
-	if (hdir == INVALID_HANDLE_VALUE)
+	if (hdir == INVALID_HANDLE_VALUE) {
+		log_err(-1, __func__, "failed in FindFirstFile");
 		return NULL;
-
+	}
 	dir = (DIR *)malloc(sizeof(DIR));
 
-	if (dir == NULL)
+	if (dir == NULL) {
+		log_err(errno, __func__, "failed to allocate memory for dir");
```

Comment:
Did you miss adding errno = GetLastError(); before this line?. errno is not defined in this function won't this throw an error here. or am i missing something?

Reply:
no its intentional. malloc sets the errno